nodejs/eks/nodegroup.ts

```diff
@@ -1800,6 +1800,9 @@ Content-Type: text/x-shellscript; charset="us-ascii"
         {
             userData,
             metadataOptions,
+            // We need to always supply an imageId, otherwise AWS will attempt to merge the user data which will result in
+            // nodes failing to join the cluster.
+            imageId: getRecommendedAMI(args, core.cluster.version, parent),
```

Comment:
Do we want to enable overriding the AMI here?
Auto upgrading nodes might not be something users want to do

Reply:
That makes sense. WDYT about exposing that in another PR to reduce scope on this PR?